processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -105,79 +103,40 @@ public static void checkInterrupted()
   }
 
   public static final String QUERYID = "queryId";
-  private final DataSource dataSource;
+
   private final boolean descending;
   private final Map<String, Object> context;
-  private final QuerySegmentSpec querySegmentSpec;
-  private volatile Duration duration;
 
   public BaseQuery(
-      DataSource dataSource,
```

Comment:
It will break source compatibility anyway.

Instead of changing `BaseQuery` and adding `SingleSourceBaseQuery`, maybe leave `BaseQuery` compatible and add "MultiSourceBaseQuery"?